lib/state_machine/src/ChargerStateMachine.cpp

```diff
@@ -0,0 +1,111 @@
+#include "ChargerStateMachine.h"
+
+
+//logic for changing states - still need to account for dial_position
+ChargerState_e ChargerStateMachine::evaluate_charger_state_machine(ChargerState_e _current_state, bool balancing_enabled, int dial_position, bool start_button_pressed) 
```

Comment:
I don't think that _current_state should be an argument to the function. You can just reference the member variable.